src/VisualStudio/CSharp/Impl/Options/IntelliSenseOptionPageControl.xaml.cs

```diff
@@ -23,22 +23,22 @@ public IntelliSenseOptionPageControl(OptionStore optionStore) : base(optionStore
             BindToOption(Show_completion_item_filters, CompletionViewOptions.ShowCompletionItemFilters, LanguageNames.CSharp);
             BindToOption(Highlight_matching_portions_of_completion_list_items, CompletionViewOptions.HighlightMatchingPortionsOfCompletionListItems, LanguageNames.CSharp);
 
-            BindToOption(Show_completion_list_after_a_character_is_typed, CompletionOptions.Metadata.TriggerOnTypingLetters, LanguageNames.CSharp);
-            Show_completion_list_after_a_character_is_deleted.IsChecked = this.OptionStore.GetOption(CompletionOptions.Metadata.TriggerOnDeletion, LanguageNames.CSharp) == true;
+            BindToOption(Show_completion_list_after_a_character_is_typed, CompletionOptionsMetadata.TriggerOnTypingLetters, LanguageNames.CSharp);
+            Show_completion_list_after_a_character_is_deleted.IsChecked = this.OptionStore.GetOption(CompletionOptionsMetadata.TriggerOnDeletion, LanguageNames.CSharp) == true;
             Show_completion_list_after_a_character_is_deleted.IsEnabled = Show_completion_list_after_a_character_is_typed.IsChecked == true;
 
-            BindToOption(Never_include_snippets, CompletionOptions.Metadata.SnippetsBehavior, SnippetsRule.NeverInclude, LanguageNames.CSharp);
-            BindToOption(Always_include_snippets, CompletionOptions.Metadata.SnippetsBehavior, SnippetsRule.AlwaysInclude, LanguageNames.CSharp);
-            BindToOption(Include_snippets_when_question_Tab_is_typed_after_an_identifier, CompletionOptions.Metadata.SnippetsBehavior, SnippetsRule.IncludeAfterTypingIdentifierQuestionTab, LanguageNames.CSharp);
+            BindToOption(Never_include_snippets, CompletionOptionsMetadata.SnippetsBehavior, SnippetsRule.NeverInclude, LanguageNames.CSharp);
```

Comment:
this would also be much clearer conceptuially as CompletionsOptionsStorage.  Oh... of course this code is binding the UI feature to the actual read/write storage for that option.  etc. etc. etc.

Reply:
Makes sense.